lib/temporal/connection/grpc.rb

```diff
@@ -409,16 +414,33 @@ def signal_with_start_workflow_execution(
         client.signal_with_start_workflow_execution(request)
       end
 
-      def reset_workflow_execution(namespace:, workflow_id:, run_id:, reason:, workflow_task_event_id:)
+      def reset_workflow_execution(namespace:, workflow_id:, run_id:, reason:, workflow_task_event_id:, request_id: nil, reset_reapply_type: nil)
+        if request_id.nil?
```

Comment:
I'd put this in the layer above since the GRPC/connection layer is mostly used for mapping payloads

Reply:
I think when I was deciding where to put this behavior, I wasn't sure if the `Temporal::Connection::GRPC.*` methods were considered a part of the public API, so I didn't want to add new required arguments to it.

However, with the fact that it's broken on Temporal Server versions since v1.14, I feel like adding this as a required arguments may actually be a good thing, because it reflects that new server versions also require it. It's also possible that this method isn't considered a part of the public API of the gem, so maybe it doesn't matter.

Reply:
I don't believe it's intended to be used directly. Since the argument is required by Temporal server, making it required at this level seems to make sense too.

Reply:
I don't believe it's intended to be used directly. Since the argument is required by Temporal server, making it required at this level seems to make sense too.